wallet-new/src/Cardano/Wallet/API/Response.hs

```diff
@@ -225,3 +230,55 @@ instance Accept ValidJSON where
 
 instance ToJSON a => MimeRender ValidJSON a where
     mimeRender _ = mimeRender (Proxy @ JSON)
+
+
+--
+-- Error from parsing / validating JSON inputs
+--
+
+newtype JSONValidationError
+    = JSONValidationFailed Text
+    deriving (Generic, Show, Eq)
+
+deriveGeneric ''JSONValidationError
+
+instance ToJSON JSONValidationError where
+    toEncoding (JSONValidationFailed weValidationError) = pairs $ mconcat
+        [ "message"    .= String "JSONValidationFailed"
+        , "status"     .= String "error"
+        , "diagnostic" .= object
+            [ "validationError" .= weValidationError
+            ]
+        ]
+
+instance FromJSON JSONValidationError where
+    parseJSON (Object o)
+        | HMS.member "message" o =
+            case HMS.lookup "message" o of
+                Just "JSONValidationFailed" ->
+                    JSONValidationFailed <$> ((o .: "diagnostic") >>= (.: "validationError"))
+                _ ->
+                    fail "Incorrect JSON encoding for JSONValidationError"
+
+        | otherwise =
+            fail "Incorrect JSON encoding for JSONValidationError"
+
+    parseJSON invalid =
+        typeMismatch "JSONValidationError" invalid
```

Comment:
No usage of generics here, to have consistent approach with the work recently done on Errors.

Reply:
Note that `withObject` does the pattern match/note dance for you